lib/CodeBuilder/Domain/Builder/ConstructorBuilder.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Domain\Builder;
+
+class ConstructorBuilder extends MethodBuilder
+{
+    /**
+     * @param mixed $originalType
+     */
+    public function returnType(string $returnType, $originalType = null): MethodBuilder
+    {
+        trigger_error('You can not have a return type on a constructor', E_USER_NOTICE);
```

Comment:
throw an exception here - it's a developer error not a user error?

Reply:
but equally better to not have this method at all. 

for the sake of simplicity we could simply forget the semantics of `__construct` and add visibility on the `parameter`.

Reply:
(for example, look at `tolerant-parser/Node/Parameter` which has a visiblity token.